devices/System.Buffers.Binary.BinaryPrimitives/BinaryPrimitives.cs

```diff
@@ -206,6 +206,52 @@ public static ulong ReadUInt64LittleEndian(SpanByte source)
                 (ulong)source[3] << 24 | (ulong)source[2] << 16 | (ulong)source[1] << 8 | (ulong)source[0]);
         }
 
+        /// <summary>
+        /// Reads a <see cref="float"/> from the beginning of a read-only span of bytes, as big endian.
+        /// </summary>
+        /// <param name="source">The read-only span to read.</param>
+        /// <returns>The big endian value.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">source is too small to contain a <see cref="float"/>.</exception>
+        public static float ReadSingleBigEndian(SpanByte source)
+        {
+            if (source.Length < 4)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            uint ieee754_bits = ReadUInt32BigEndian(source);
+            float flt;
+            unsafe
```

Comment:
nice usage of unsafe.
@josesimoes just to double check, bindings using the nuget won't have to compile unsafe to use this? If so, then I would prefer a way that is safe if possible. Because that may be quite a constraint if we had to change always for unsafe.

Reply:
Not 100% sure on this, but I think this is "contained" to the assembly being compiled. The one referencing it are unaware of this.

Reply:
That's why I asked :-D Because I was not 100% sure neither ;-)

Reply:
You already have a bunch of projects in `nanoFramework.IoT.Device` that require unsafe code. One example is `nanoFramework.System.Numerics`. When you add that as reference to your project, your own project does not require to allow unsafe either. A bunch of other device packages also use unsafe. So this shouldn't be an issue.